osclib/adi_command.py

```diff
@@ -87,9 +87,13 @@ def create_new_adi(self, wanted_requests, by_dp=False, split=False):
             print(Fore.YELLOW + (group if group != '' else 'wanted') + Fore.RESET)
 
             name = None
+            nonfree_repo_required = False
             for request in splitter.grouped[group]['requests']:
                 request_id = int(request.get('id'))
                 target_package = request.find('./action/target').get('package')
+                target_project = request.find('./action/target').get('project')
+                if self.api.cnonfree and self.api.cnonfree == target_project:
```

Comment:
This condition is in both here and in `RequestSplitter`. Should be able to get away with only one rather than duplicate check.